modules.yml

```diff
@@ -1065,3 +1065,19 @@
   path: reporting/xml
   required_keys: []
   version: '1.1'
+- author: Joe Gray (@C_3PJoe)
+  dependencies:
+  - pandas
```

Comment:
What is `pandas`?

Also, you don't need to list deps that are Python builtins. Recon-ng has a built in wrapper for requests that must be used for requests, so see the documentation on module building in the Recon-ng wiki.

If you want to use BS4 you can, but know that it is an extra dependency that people will have to install. lxml is a powerful parser and is already installed with Recon-ng if you would prefer to use that.

Reply:
Thanks for the feedback. I am relatively new to writing code beyond editing files from time to time or making slight modifications. 

Pandas is a package that allows in-memory data manipulation and analysis via DataFrames (which look kind of like an Excel spreadsheet or a csv). I was introduced to it when tinkering with some Data Science projects and fell in love. I am open to using another method, I only used pandas because of the ease in working with the data.

RE BS4, I just found that lxml has a SoupParser, which I will modify the code to use in favor of the additional lift of using BS4.

Reply:
I found a way to remove the redundant use of pandas, but was unable to cleanly parse the HTML using LXML, so I went back to using BS4. Validated that it works and added a REGEX/substitution to remove brackets from email addresses when writing to the db.

Reply:
Replacing BeautifulSoup here is very simple as all it requires is parsing the HTML with the etree parser and replicating `'div', {'class': 'result'}` with an XPath query. In fact, I would argue that this entire module could be made MUCH more resilient by replacing the regular expressions with XPath queries too. If you aren't familiar with XPath, it's work learning, and other modules and the core codebase include examples of how to use lxml to do this. Once you learn it, you'll see just how powerful it can be vs. BS. In fact, BS uses lxml under the hood as its primary parsing engine. All you are doing is wrapping lxml with BS. It will be better all around, for users and you, to replace BS with lxml.

Reply:
Thanks for the feedback! I will use today as a time to learn XPath and see what I can concoct. I am always down to learn something new. I appreciate your patience and really appreciate your commitment to the community with this tool.

Reply:
Hi Tim,

I fought with lxml and XPath all day to no avail. I really struggled with finding a cohesive set of documentation that provides insight to implement it. I am not one to quit, much less rage quit, but the whole experience of trying to wrap my head around XPath seriously made me consider rage quitting. I reached out to people on various slacks, Twitter, and Facebook. I wasn't able to get much traction from those sources. 

I think, for now, let's leave BS4 in the code and when the world isn't upside down and I have a moment to breathe, I will re-examine XPath for inclusion on a later revision. If I didn't have a few other projects on my plate, I would be more apt to continue the good fight right now, but combined with cabin fever and an increased workload from COVID and other obligations, I can't sink the time required into picking it up right now.

Reply:
Hi, I hope you had a good Easter weekend. I am just circling back around to see if any other action is required for this revision.

Reply:
I'm not an approver on this, but I would say the data manipulation you're doing here could be done in a list of dictionaries. Recon errs on the side of fewer external dependencies, and I don't think the ease of use with pandas doesn't justify adding it here.

Reply:
Thanks @cam-barts. I have since revised the code to remove pandas. The only external dependency to install now is BS4. I have committed that 5 days ago. Do I need to close this pull and submit another?

Reply:
Sorry I'll see about testing and approving this later today.

Reply:
TYVM

Reply:
@jocephus Just push to this same branch it the changes should update the PR.

@MrJester Before approving, see how hard it would be porting BS4 to lxml. At first glance, looks like a super simple change. One Xpath would replace the need for the entire BS4 library, which incidentally is what BS4 is already using under the hood.